demo/site/src/pages/preview/admin/page.tsx

```diff
@@ -1,10 +1,33 @@
 import { BlockPreviewProvider, IFrameBridgeProvider, useIFrameBridge } from "@comet/cms-site";
+import { PageContentBlockData } from "@src/blocks.generated";
 import { PageContentBlock } from "@src/blocks/PageContentBlock";
+import { recursivelyLoadBlockData } from "@src/recursivelyLoadBlockData";
+import createGraphQLClient from "@src/util/createGraphQLClient";
 import * as React from "react";
 
 const PreviewPage: React.FunctionComponent = () => {
     const iFrameBridge = useIFrameBridge();
-    return <div>{iFrameBridge.block && <PageContentBlock data={iFrameBridge.block} />}</div>;
+    const clientRef = React.useRef(
+        createGraphQLClient({
+            previewDamUrls: true,
+            includeInvisibleBlocks: false,
+            includeInvisiblePages: false,
+        }),
+    );
```

Comment:
Why do you need a ref for this? You could also declare the client outside of the component. Also, should the options of the client be adapted according the current settings in the admin?